### PR TITLE
Add GH pages action to fix file permissions

### PR DESCRIPTION
**Pull request by vbharadwaj-bk**
# Pull Request Checklist

Resolves: #3244 

- [x] Ensured **tests pass** and (if applicable) updated functional test output
- [x] Conformed to **code style guidelines** by running appropriate linting tools
- [ ] Added **tests** for changed code (verified by building a test site and checking the GH pages workflow)
- [ ] Updated **documentation** for changed code (N/A)

Adds a Github action that fixes file permissions before uploading (such as cache files created by Pelican Webassets if the cache is not disabled). Raises a warning for each file without proper permissions.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Release type: patch
```
Comment (avaris) [inline RELEASE.md:1]: I don't think there is need for a release. This workflow lives only on github.

Comment (justinmayer) [reply RELEASE.md:1]: I came here to say the same thing and agree that the release file should be removed from this PR.

Comment (vbharadwaj-bk) [reply RELEASE.md]: Understood, file removed.

**Comment by seanh** [comment]
This fix seems fine to me 👍 If upload-pages-artefact chokes on certain file permissions in the output directory, I see no reason why the workflow shouldn't fix those permissions to ensure that doesn't happen

**Review by avaris — APPROVED** [review]

**Review by justinmayer — APPROVED** [review]
Thanks to @vbharadwaj-bk for the enhancement and to @avaris and @seanh for reviewing. 🎖️

**Pull request merged**